packages/core/src/render3/node_selector_matcher.ts

```diff
@@ -288,3 +267,26 @@ export function matchingProjectionSelectorIndex(
   }
   return 0;
 }
+
+function getNameOnlyMarkerIndex(nodeAttrs: TAttributes) {
+  let nameOnlyMarkerIdx = nodeAttrs && nodeAttrs.length;
+  for (let i = 0; i < nodeAttrs.length; i++) {
+    const nodeAttr = nodeAttrs[i];
+    if (isNameOnlyAttributeMarker(nodeAttr)) {
+      nameOnlyMarkerIdx = i;
+      break;
+    }
+  }
+  return nameOnlyMarkerIdx;
+}
+
+function matchTemplateAttribute(attrs: TAttributes, name: string): number {
+  let i = attrs.indexOf(AttributeMarker.Template);
+  if (i > -1 && i < attrs.length) {
```

Comment:
Looks like the `do-while` could become a `while`, as the do-block is guarded within the exact same condition as the while guard:

```suggestion
  if (i !== -1) {
    while (i < attrs.length) {
      i++;
      if (attrs[i] === name) return i;
    }
  }
```

Reply:
It's not quite trivial but I think that we can do:

```ts
function matchTemplateAttribute(attrs: TAttributes, name: string): number {
  let i = attrs.indexOf(AttributeMarker.Template);
  if (i > -1) {
    i++;
    while (i < attrs.length) {
      if (attrs[i] === name) return i;
      i++;
    }
  }
  return -1;
}
```

Reply:
Oh I misjudged the `i++`! Looking at your proposal, we could replace the 2 `i++`'s with `while (++i < attrs.length)` (or would it then become *clever code* 🙄)

Reply:
I'm not going there :-P